### PR TITLE
Name field var_CD of vehicle struct. Refactor to enum

### PR DESCRIPTION
**Pull request by Gymnasiast**
Continuation of https://github.com/OpenRCT2/OpenRCT2/pull/10346.

### CODE REVIEW

File: src/openrct2/ride/CableLift.cpp [hunk 1]
```diff
@@ -236,8 +236,8 @@ static bool sub_6DF01A_loop(rct_vehicle* vehicle)
 
         uint16_t trackProgress = vehicle->track_progress + 1;
 
-        const rct_vehicle_info* moveInfo = vehicle_get_move_info(vehicle->var_CD, vehicle->track_type, 0);
-        uint16_t trackTotalProgress = vehicle_get_move_info_size(vehicle->var_CD, vehicle->track_type);
+        const rct_vehicle_info* moveInfo = vehicle_get_move_info(vehicle->TrackSubposition, vehicle->track_type, 0);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/CableLift.cpp]: Modify `vehicle_get_move_info` to change the variable names to `trackSubPosition`

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -763,73 +763,73 @@ static const struct
...
-    if (offset >= gTrackVehicleInfo[cd][typeAndDirection]->size)
+    if (offset >= gTrackVehicleInfo[trackSubposition][typeAndDirection]->size)
     {
         return false;
     }
     return true;
 }
 
-const rct_vehicle_info* vehicle_get_move_info(int32_t cd, int32_t typeAndDirection, int32_t offset)
+const rct_vehicle_info* vehicle_get_move_info(int32_t track_status, int32_t typeAndDirection, int32_t offset)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: Modify `vehicle_get_move_info` to change the variable names to `trackSubPosition`

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -763,73 +763,73 @@ static const struct
...
     {
         static constexpr const rct_vehicle_info zero = {};
         return &zero;
     }
-    return &gTrackVehicleInfo[cd][typeAndDirection]->info[offset];
+    return &gTrackVehicleInfo[track_status][typeAndDirection]->info[offset];
 }
 
-uint16_t vehicle_get_move_info_size(int32_t cd, int32_t typeAndDirection)
+uint16_t vehicle_get_move_info_size(int32_t track_status, int32_t typeAndDirection)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: As above

File: src/openrct2/ride/Vehicle.h [hunk 4]
```diff
@@ -150,6 +150,28 @@ enum VEHICLE_STATUS
 
 struct rct_vehicle_sound_params;
 
+enum VEHICLE_TRACK_SUBPOSITION : uint8_t
+{
+    VEHICLE_TRACK_SUBPOSITION_0,
+    VEHICLE_TRACK_SUBPOSITION_CHAIRLIFT_1,
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.h]: 1 represents the going out (left side) cable, 2 represents the return (right side) cable, 3 is the end or far bullwheel, 4 is the start or close bullwheel. We can probably make some better names based off that knowledge.

File: src/openrct2/ride/Vehicle.h [hunk 5]
```diff
@@ -150,6 +150,28 @@ enum VEHICLE_STATUS
...
+    VEHICLE_TRACK_SUBPOSITION_CHAIRLIFT_2,
+    VEHICLE_TRACK_SUBPOSITION_CHAIRLIFT_END_3,
+    VEHICLE_TRACK_SUBPOSITION_CHAIRLIFT_END_4,
+    VEHICLE_TRACK_SUBPOSITION_GO_KART_5,
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.h]: 5 left track, 6 right track, 7 Move from left to right, 8 Move from right to left

File: src/openrct2/ride/Vehicle.cpp [hunk 6]
```diff
@@ -7490,7 +7490,7 @@ static void loc_6DB481(rct_vehicle* vehicle)
     }
     if ((scenario_rand() & 0xFFFF) <= probability)
     {
-        vehicle->var_CD += 2;
+        vehicle->TrackSubposition += VEHICLE_TRACK_SUBPOSITION_GO_KART_7 - VEHICLE_TRACK_SUBPOSITION_GO_KART_5;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: Add a comment that this is switching lanes from left to right or vice versa.

File: src/openrct2/ride/Vehicle.cpp [hunk 7]
```diff
@@ -7490,7 +7490,7 @@ static void loc_6DB481(rct_vehicle* vehicle)
     }
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: This function can be named `VehicleUpdateGoKartAttemptSwitchLanes`

File: src/openrct2/ride/Vehicle.cpp [hunk 8]
```diff
@@ -7919,7 +7920,7 @@ static void sub_6DBF3E(rct_vehicle* vehicle)
         }
         if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_GO_KART)
         {
-            cx = vehicle->var_CD == 6 ? 18 : 20;
+            cx = vehicle->TrackSubposition == VEHICLE_TRACK_SUBPOSITION_GO_KART_6 ? 18 : 20;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: This is determining the stop positions for gokarts. If in left lane its further along the track than the right lane.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp]: Its not possible to overtake in station elements thats why it doesn't check for overtake states (7 and 8)

File: src/openrct2/ride/Vehicle.h [hunk 9]
```diff
@@ -150,6 +150,28 @@ enum VEHICLE_STATUS
...
+    VEHICLE_TRACK_SUBPOSITION_GO_KART_7,
+    VEHICLE_TRACK_SUBPOSITION_GO_KART_8,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_START_9 = 9,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_PATH_A_9 = 9,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_BALL_PATH_A_10,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_PATH_B_11,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_BALL_PATH_B_12,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_PATH_C_13,
+    VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_BALL_PATH_C_14,
+    VEHICLE_TRACK_SUBPOSITION_REVERSER_RC_15,
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.h]: 15 is Rear bogie, 16 is front bogie I can't quite confirm which way round they are though as it doesnt make 100% sense.

File: src/openrct2/ride/TrackData.cpp [hunk 10]
```diff
@@ -29699,23 +29699,23 @@ static constexpr const rct_vehicle_info_list *TrackVehicleInfoList_8BAD28[] = {
...
-    TrackVehicleInfoList_9334D0,
-    TrackVehicleInfoList_9341B0,
-    TrackVehicleInfoList_934E90,
-    TrackVehicleInfoList_935B70,
-    TrackVehicleInfoList_936850,
-    TrackVehicleInfoList_937530,
-    TrackVehicleInfoList_8B9F98,
-    TrackVehicleInfoList_8BAD28,
+    TrackVehicleInfoList_8B8F98,// VEHICLE_TRACK_SUBPOSITION_0
+    TrackVehicleInfoList_8BBAB8,// VEHICLE_TRACK_SUBPOSITION_CHAIRLIFT_1
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackData.cpp]: Update these comments and this is good to go

**Comment by Gymnasiast** [comment]
Note: do a merge, not a squash, so that the original author can be saved.

**Pull request merged**